core/domain/story_domain.py

```diff
@@ -0,0 +1,359 @@
+        """
+        self.id = story_id
+        self.title = title
+        self.description = description
+        self.notes = notes
+        self.topic = topic
+        self.story_nodes = story_nodes
+        self.schema_version = schema_version
+        self.language_code = language_code
+        self.created_on = created_on
+        self.last_updated = last_updated
+        self.version = version
+
+    def to_dict(self):
+        """Returns a dict representing this Story domain object.
+
+        Returns:
+            A dict, mapping all fields of Story instance.
+        """
+        return {
+            'id': self.id,
+            'title': self.title,
+            'description': self.description,
+            'notes': self.notes,
+            'language_code': self.language_code,
+            'topic': self.topic,
+            'schema_version': self.schema_version,
+            'version': self.version,
+            'story_nodes': [
+                node.to_dict() for node in self.story_nodes
+            ]
+        }
+
+    @classmethod
+    def from_dict(
+            cls, story_dict, story_version=0,
+            story_created_on=None, story_last_updated=None):
+        """Return a Story domain object from a dict.
+
+        Args:
+            story_dict: dict. The dictionary representation of  the
+                story.
+            story_version: int. The version of the story.
+            story_created_on: datetime.datetime. Date and time when the
+                story is created.
+            story_last_updated: datetime.datetime. Date and time when
+                the story is updated last time.
+
+        Returns:
+            Story. The corresponding Story domain object.
+        """
+        story = cls(
+            story_dict['id'], story_dict['title'],
+            story_dict['description'], story_dict['notes'],
+            story_dict['topic'],
+            [
+                StoryNode.from_dict(node_dict)
+                for node_dict in story_dict['story_nodes']
+            ], story_dict['schema_version'], story_dict['language_code'],
+            story_version, story_created_on, story_last_updated)
+
+        return story
+
+    @classmethod
+    def create_default_story(
+            cls, story_id, title=feconf.DEFAULT_STORY_TITLE,
+            description=feconf.DEFAULT_STORY_DESCRIPTION,
+            notes=feconf.DEFAULT_STORY_NOTES,
+            language_code=constants.DEFAULT_LANGUAGE_CODE):
+        """Returns a story domain object with default values.
+
+        Args:
+            story_id: str. The unique id of the story.
+            title: str. The title of the story.
+            category: str. The category of the story.
+            objective: str. The objective of the story.
+            language_code: str. The language code of the story (like 'en'
+                for English).
+
+        Returns:
+            story. The story domain object with the default
```

Comment:
Type should be Story, not story. (Use the output when you do `type(X)` in Python.)

Reply:
Done.